### PR TITLE
Bug fix

### PR DESCRIPTION
**Pull request by provishalk**

### CODE REVIEW

File: src/components/Posts/AddPost/AddPost.js [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
+import { useHistory } from "react-router-dom";
+import Spinner from 'react-bootstrap/Spinner';
 const AddPost = () => {
+  let history = useHistory();
   const user = JSON.parse(localStorage.getItem("user"));
   const [text, setText] = useState("");
+  const [isButtonClicked, setIsButtonClicked] = useState(false);
   const onPostClickHandler = (event) => {
     event.target.disabled = true;
+    setIsButtonClicked(true);
```
Comment (shivdotpy) [inline src/components/Posts/AddPost/AddPost.js]: use other variable name, such as isLoading (self explanatory)

File: src/components/Posts/AddPost/AddPost.js [hunk 2]
```diff
@@ -1,16 +1,20 @@
...
-import { CREATE_POST } from "../../../utils/constants";
+import { useHistory } from "react-router-dom";
+import Spinner from 'react-bootstrap/Spinner';
 const AddPost = () => {
+  let history = useHistory();
   const user = JSON.parse(localStorage.getItem("user"));
   const [text, setText] = useState("");
+  const [isButtonClicked, setIsButtonClicked] = useState(false);
   const onPostClickHandler = (event) => {
     event.target.disabled = true;
```
Comment (shivdotpy) [inline src/components/Posts/AddPost/AddPost.js:14]: Don't use like this, use isLoading variable to disable the button

File: src/components/Posts/Post/Post.js [hunk 3]
```diff
@@ -11,11 +11,11 @@ import { AiOutlineHeart } from 'react-icons/ai';
 const Post = ({ name, text, createdAt, likes, postId, postedBy, comments }) => {
   const [open, setOpen] = useState(false);
   const user = JSON.parse(localStorage.getItem("user"));
-  const [postLikedByCurrentUser, setPostLikedByCurrentUser] = useState(likes.includes(user._id) ? <FcLike key={1} /> : <AiOutlineHeart key={2} />)
+  const [postLikedByCurrentUser, setPostLikedByCurrentUser] = useState(likes.includes(user?._id) ? <FcLike key={1} /> : <AiOutlineHeart key={2} />)
   const onHandleLike = (event) => {
     setPostLikedByCurrentUser(postLikedByCurrentUser.key === "1" ? <AiOutlineHeart key={2} /> : <FcLike key={1} />);
     const config = {
-      headers: { Authorization: `Bearer ${user.token}` },
+      headers: { Authorization: `Bearer ${user?.token}` },
```
Comment (shivdotpy) [inline src/components/Posts/Post/Post.js]: Search for axios instance, so you can add headers in one place only and that instance can be used for URL and data only. other things can be done there

File: src/containers/Home/Home.js [hunk 4]
```diff
@@ -8,9 +8,14 @@ import io from "socket.io-client";
 import _ from "lodash";
 import alertify from "alertifyjs";
 import { OverlayTrigger, Button, Tooltip } from "react-bootstrap";
+import { FiLogOut } from 'react-icons/fi';
 
 const Home = ({ history }) => {
   const [posts, setPosts] = useState([]);
+  const isLoggedIn = localStorage.getItem("user");
```
Comment (shivdotpy) [inline src/containers/Home/Home.js]: make a util function which will return boolean is user is logged in or not

File: src/containers/Home/Home.js [hunk 5]
```diff
@@ -113,16 +118,16 @@ const Home = ({ history }) => {
               );
             })}
           </div>
-          <div className="col-0 col-md-3 col-xl-4 d-flex home__logout">
+          <div className="col-0 col-md-3 col-xl-3 d-flex home__logout">
             <div className="ml-auto">
               <OverlayTrigger
                 placement="bottom"
                 overlay={
                   <Tooltip>Logout</Tooltip>
```
Comment (shivdotpy) [inline src/containers/Home/Home.js]: Add this to label

File: src/containers/Home/Home.scss [hunk 6]
```diff
@@ -1,14 +1,25 @@
...
+  &__posts-container {
+    margin-left: auto;
+    padding-left: 100px !important;
+    padding-right: 100px !important;
+  }
+}
+@media only screen and (max-width: 1300px) {
+  .home {
+    &__posts-container {
+      padding-left: 0px !important;
```
Comment (shivdotpy) [inline src/containers/Home/Home.scss:20]: try not to use !important (not recommended)

File: src/components/Posts/AddPost/AddPost.js (hunk 1 above)
Comment (provishalk) [reply src/components/Posts/AddPost/AddPost.js]: Done

File: src/components/Posts/AddPost/AddPost.js (hunk 2 above)
Comment (provishalk) [reply src/components/Posts/AddPost/AddPost.js:14]: done

File: src/containers/Home/Home.js (hunk 5 above)
Comment (provishalk) [reply src/containers/Home/Home.js]: Done

File: src/components/Posts/Post/Post.js (hunk 3 above)
Comment (provishalk) [reply src/components/Posts/Post/Post.js]: Done

File: src/containers/Home/Home.scss (hunk 6 above)
Comment (provishalk) [reply src/containers/Home/Home.scss:20]: without !important bootstrap not allowing me to add padding. what to do?

File: src/containers/Home/Home.js (hunk 4 above)
Comment (provishalk) [reply src/containers/Home/Home.js]: Done

**Review by shivdotpy — APPROVED** [review]

**Pull request merged**